File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(x, y) {
    return [최대공약수(x, y), 최소공배수(x, y)];
}

const 최대공약수 = (x, y) => {
    while(1){
        if(x % y === 0 ) break;
        [x, y] = [y, x % y];
    }
    return y;
}

const 최소공배수 = (x, y) => {
    return x*y / 최대공약수(x, y);
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
import math

def lcm(n, m):
    if n == m:
        return n
    
    if n < m:
        min_v = n
        max_v = m
    else:
        min_v = n
        max_v = m
        
    i = 1
    temp = 0
    while(True):
        temp = min_v * i
        
        if temp == max_v:
            return min_v
        
        elif temp < max_v:
            if max_v % temp == 0:
                return max_v
            
        else:
            if temp % max_v == 0:
                return temp
            
        i += 1
        
def solution(n, m):
    return [math.gcd(n, m), lcm(n, m)]

